ruby_activerecord/boot.rb

```diff
@@ -1,8 +1,101 @@
 require './config/initializers/database'
 require './app/models/shop'
+require './app/models/category'
+require './app/models/feedback'
 require './app/models/product'
+require './app/models/question'
+require './app/models/tag'
 
-shop = Shop.find(1)
+#shop = Shop.new(
+#  domain: 'rozetka.com.ua',
+#  name: 'Rozetka',
+#  address: 'Lviv',
+#  is_open: true,
+#  staff_count: 500
+#)
+#shop.save
+#
+#category1 = Category.new(title: 'boots')
+#category1.save
+#category2 = Category.new(title: 'jackets')
+#category2.save
+#
+#tag1 = Tag.create(title: 'winter')
+#product1 = Product.create(title: 'winter jacket1', price: '3500', shop: shop, category: category2)
+#
+#first_question = Question.new(
+#  comment: "47th size is available?",
+#  username: 'Maksym',
+#  product: product1
+#)
+#first_question.save
+#
+#first_feedback = Feedback.new(rate: 4, comment: 'very warm jacket', product: product1)
+#first_feedback.save
+#
+#product2 = Product.create(title: 'winter jacket2', price: '1500', shop: shop, category: category2)
+#product3 = Product.create(title: 'winter jacket3', price: '7500', shop: shop, category: category2)
+#product4 = Product.create(title: 'winter jacket4', price: '1030', shop: shop, category: category2)
+#
+#product1.tags << tag1
+#product2.tags << tag1
+#product3.tags << tag1
+#product4.tags << tag1
+
+#product3 = Product.find(3)
+#first_question = Question.new(
+#  comment: "It`s original?`",
+#  product: product3
+#)
+#first_question.save
+
+def sort_tags_products_by_price(tag_name)
+  tag = Tag.find_by(title: tag_name)
+  tag.products.order(:price).each { | p | puts "#{p.title} price is: #{p.price}" }
+end
+
+def products_in_shop_with_category(category_name, shop_name)
+  category = Category.find_by(title: category_name)
+  shop = Shop.where(name: shop_name)
```

Comment:
best flow in your case should be:
`get shop -> get category by shop -> get products by category`
1. More logical structure is 'Product belongs to Category, Category belongs to shop'
And with this structure, if you don't need to execute shop and category directly, you can use SQL operator JOIN
```ruby
products = Product.joins(category: :shop).where(shops: { name: shop_name })
products.each { | p | puts "#{shop_name} has a category - #{category_name}: #{p.title}" }
```